### PR TITLE
Tweet Message Scoring

### PR DESCRIPTION
**Pull request by thekeele**

### CODE REVIEW

File: assets/js/socket.js [hunk 1]
```diff
@@ -56,41 +56,87 @@ socket.connect()
...
-  console.log("payload", payload)
+// multi-column support
+let messageContainers = document.querySelectorAll("[id*='message_column']")
+let n_columns = messageContainers.length;
 
-  let tileContainer = document.createElement("div")
-  tileContainer.className = "tile is-ancestor"
+// color tweet depending on score with five classes
+function score_color(score) {
+  if (score < -0.75) {
```
Comment (thekeele) [inline assets/js/socket.js]: Need to handle the case of undefined score.

File: assets/js/socket.js [hunk 2]
```diff
@@ -56,41 +56,87 @@ socket.connect()
...
+  } else {
+    return "#23d160"; //very positive is-success
+  }
+}
 
-  let tweetTile = document.createElement("div")
-  tweetTile.className = "tile is-parent"
+// return emoji depending on score
+function score_emoji(score) {
+  if (score < -0.75) {
```
Comment (thekeele) [inline assets/js/socket.js]: Same as above.

File: assets/js/socket.js [hunk 3]
```diff
@@ -56,41 +56,87 @@ socket.connect()
...
+}
 
-  let tweetNotification = document.createElement("article")
-  if(payload.sentiment == "positive") {
-    tweetNotification.className = "tile is-child notification is-success"
-  } else if (payload.sentiment == "negative") {
-    tweetNotification.className = "tile is-child notification is-danger"
+// pretty print for tweet score
+function format_sentiment(score) {
+  if (score < 0) {
```
Comment (thekeele) [inline assets/js/socket.js]: Same as above.

File: lib/sen_tweet_web/templates/page/index.html.eex [hunk 4]
```diff
@@ -1 +1,56 @@
-<div id="messages"></div>
+<div id="messages-container">
+
+  <!-- Tweet template -->
+  <article id="tweet-message" class="message" style="display: none;">
+    <div id="header" class="message-header">
+      <p id="sentiment"></p>
+      <!--<p id="score"></p>-->
```
Comment (thekeele) [inline lib/sen_tweet_web/templates/page/index.html.eex]: remove?

File: assets/js/socket.js [hunk 5]
```diff
@@ -56,41 +56,87 @@ socket.connect()
...
-  tweetNotification.appendChild(sentiment)
-  tweetNotification.appendChild(text)
+  new_tweet.querySelector("#tweet-text").innerText = payload.text;
 
-  tweetTile.appendChild(tweetNotification)
-  tileContainer.appendChild(tweetTile)
+  new_tweet.querySelector("#reply-count").innerText = payload.reply_count;
+  new_tweet.querySelector("#retweet-count").innerText = payload.retweet_count;
+  new_tweet.querySelector("#like-count").innerText = payload.favorite_count;
+  console.log(new_tweet);
```
Comment (thekeele) [inline assets/js/socket.js]: remove?

File: assets/js/socket.js [hunk 6]
```diff
@@ -56,41 +56,87 @@ socket.connect()
...
+channel.on("new_tweet", payload => {
+  console.log("payload", payload)
+
+  var tweet_message = document.getElementById("tweet-message");
+  var new_tweet = tweet_message.cloneNode(true);
+  new_tweet.style = null;
+
+  new_tweet.querySelector("#header").style.backgroundColor = score_color(payload.score);
+  new_tweet.querySelector("#sentiment").innerText = format_sentiment(payload.score);
+  //new_tweet.querySelector("#score").innerText = format_sentiment(payload.score);
```
Comment (thekeele) [inline assets/js/socket.js]: remove?

**Comment by thekeele** [comment]
Let me know what you think about the navbar.

File: lib/sen_tweet_web/templates/page/index.html.eex (hunk 4 above)
Comment (rcehemann) [reply lib/sen_tweet_web/templates/page/index.html.eex]: Overlooked some of this stuff. Regarding null scores, it should be easy enough to return "" when the score is nan but there will be boilerplate. Good use for a function wrapper.. if I can figure out how to do that in js

Comment (thekeele) [reply lib/sen_tweet_web/templates/page/index.html.eex]: Why not Elixir ? :feelsgood:

**Comment by thekeele** [comment]
Whoops, didn't notice that. Will round to next integer.

**Pull request merged**